libpermian/plugins/github/__init__.py

```diff
@@ -0,0 +1,119 @@
+import requests
+import jinja2
+import logging
+import json
+from libpermian.plugins import api
+from libpermian.reportsenders.base import BaseReportSender
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class GitHubReportingException(Exception):
+    def __init__(self, status_code, text):
+        super().__init__(f'Error {status_code}, {text}')
+
+
+@api.reportsenders.register('github-pr')
+class GitHubPullRequestReportSender(BaseReportSender):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        github_token = self.settings.get('github', 'token')
+        self.pr_id = self.settings.get('github', 'pull-request')
+        self.repository = self.settings.get('github', 'repository')
+        self.api_url = 'https://api.github.com'
+        
+        self.headers = {'Accept': 'application/vnd.github.v3+json', \
+                        'Authorization': f'token {github_token}'}
+
+        # Get output text template
+        if self.reporting.data and 'output-template' in self.reporting.data:
+            jinja_env = jinja2.Environment(loader=jinja2.BaseLoader)
+            self.template = jinja_env.from_string(self.reporting.data.get('output-template'))
+        else:
+            jinja_env = jinja2.Environment(loader=jinja2.PackageLoader('libpermian.plugins.github', 'templates'))
+            self.template = jinja_env.get_template('output_text.j2')
+
+    def render_output_text(self):
+        return self.template.render(crcs=self.caseRunConfigurations,
+                                    reportsender=self)
+
+    def get_payload(self, status="in_progress", conclusion=None, head_sha=None):
+        payload = {"name": self.testplan.name,
+                   "status": status,
+                   "output": {"title": self.testplan.name,
+                              "summary": self.testplan.description,
+                              "text": self.render_output_text()
+                             }
+                  }
+
+        if conclusion:
+            payload['conclusion'] = conclusion
+        if head_sha:
+            payload['head_sha'] = head_sha
+
+        return json.dumps(payload)
+
+    def setUp(self):
+        if self.dry_run:
+            return
+
+        # Get head SHA from pull request
+        response = requests.get(
```

Comment:
Re-using response variable can be confusing and can simply lead to error when copy-pasting code. I'd suggest adding a prefix to the variables like `pulls_response` and `checkruns_response`.